pilot/cmd/pilot-agent/status/util/stats.go

```diff
@@ -58,30 +61,35 @@ func (s *Stats) String() string {
 		s.LDSUpdatesRejection)
 }
 
-// GetServerState returns the current Envoy state by checking the "server.state" stat.
-func GetServerState(localHostAddr string, adminPort uint16) (*uint64, error) {
+// GetReadinessStats returns the current Envoy state by checking the "server.state" stat.
+func GetReadinessStats(localHostAddr string, adminPort uint16) (*uint64, uint64, error) {
 	// If the localHostAddr was not set, we use 'localhost' to void emppty host in URL.
 	if localHostAddr == "" {
 		localHostAddr = "localhost"
 	}
 
-	stats, err := doHTTPGet(fmt.Sprintf("http://%s:%d/stats?usedonly&filter=%s", localHostAddr, adminPort, statServerState))
+	stats, err := doHTTPGet(fmt.Sprintf("http://%s:%d/stats?usedonly&filter=%s", localHostAddr, adminPort, readyStatsRegex))
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 	if !strings.Contains(stats.String(), "server.state") {
-		return nil, fmt.Errorf("server.state is not yet updated: %s", stats.String())
+		return nil, 0, fmt.Errorf("server.state is not yet updated: %s", stats.String())
+	}
+
+	if !strings.Contains(stats.String(), "listener_manager.workers_started") {
+		return nil, 0, fmt.Errorf("listener_manager.workers_started is not yet updated: %s", stats.String())
 	}
 
 	s := &Stats{}
 	allStats := []*stat{
 		{name: statServerState, value: &s.ServerState},
+		{name: statWorkersStarted, value: &s.WorkersStarted},
 	}
 	if err := parseStats(stats, allStats); err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
-	return &s.ServerState, nil
+	return &s.ServerState, s.WorkersStarted, nil
```

Comment:
is this the number of workers started?

Reply:
no. It is just a gauge that indicates whether all workers have started or not